### ISSUE TITLE
Fails on sync:refresh

### ISSUE THREAD

**Issue by snize** [acquia/blt#2152]
My system information:
* Operating system type: macOS
* Operating system version: 10.12.6
* BLT version: 8.9.9

This problem is caused by re-adding `@executeInDrupalVm` in blt 8.9.7

Output of `blt doctor`:
```
+----------------------+---------------------------------------------------------------------------------+
| Property             | Value                                                                           |
+----------------------+---------------------------------------------------------------------------------+
| %paths               |                                                                                 |
|  - %root             | /var/www/myproject/docroot                                                         |
|  - %site             | sites/default                                                                   |
|  - %modules          | modules                                                                         |
|  - %themes           | themes                                                                          |
|  - %config-sync      | /var/www/myproject/config/default                                                  |
|  - %files            | sites/default/files                                                             |
|  - %private          | /var/www/myproject/files-private                                                   |
|  - %temp             | /tmp                                                                            |
| admin-theme          | seven                                                                           |
| blt-version          | 8.9.9                                                                           |
| bootstrap            | Successful                                                                      |
| config-sync          | /var/www/myproject/config/default                                                  |
| db-driver            | mysql                                                                           |
| db-hostname          | localhost                                                                       |
| db-name              | drupal                                                                          |
| db-password          | drupal                                                                          |
| db-port              | 3306                                                                            |
| db-status            | Connected                                                                       |
| db-username          | drupal                                                                          |
| dev-desktop-enabled  | false                                                                           |
| drupal-settings-file | sites/default/settings.php                                                      |
| drupal-version       | 8.3.7                                                                           |
| drupal-vm-enabled    | true                                                                            |
| drush-alias-files    | /var/www/myproject/docroot/../drush/site-aliases/aliases.drushrc.php               |
|                      | /var/www/myproject/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.php  |
|                      | /var/www/myproject/docroot/../drush/site-aliases/example.local.aliases.drushrc.php |
|                      | /var/www/myproject/docroot/../drush/site-aliases/myproject.aliases.drushrc.php        |
| drush-conf           | /var/www/myproject/drush/drushrc.php                                               |
| drush-script         | /var/www/myproject/vendor/drush/drush/drush.php                                    |
| drush-temp           | /tmp                                                                            |
| drush-version        | 8.1.15                                                                          |
| files                | sites/default/files                                                             |
| install-profile      | lightning                                                                       |
| local-drushrc        | /var/www/myproject/docroot/sites/default/local.drushrc.php                         |
| local-settings       | /var/www/myproject/docroot/sites/default/settings/local.settings.php               |
| modules              | modules                                                                         |
| php-bin              | /usr/bin/php                                                                    |
| php-conf             | /etc/php/5.6/cli/php.ini                                                        |
| php-mysql            | /var/run/mysqld/mysqld.sock                                                     |
| php-os               | Linux                                                                           |
| private              | /var/www/myproject/files-private                                                   |
| root                 | /var/www/myproject/docroot                                                         |
| shell                |                                                                                 |
| site                 | sites/default                                                                   |
| temp                 | /tmp                                                                            |
| theme                | myproject                                                                          |
| themes               | themes                                                                          |
| uri                  | http://local.myproject.jp                                                          |
| user                 |                                                                                 |
+----------------------+---------------------------------------------------------------------------------+
+--------------------------+----------------------------------------------------------------------------------+
| Check                    | Outcome                                                                          |
+--------------------------+----------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                   |
| checkCoreExists          | Drupal core exists                                                               |
| checkSettingsFile        | BLT settings are included in settings file.                                      |
| checkLocalSettingsFile   | Found your local settings file.                                                  |
| checkLocalDrushFile      | Found your local drush settings file.                                            |
| checkUri                 | $options['uri'] is set.                                                          |
| checkUriResponse         | Received a response from site http://local.myproject.jp.                            |
| checkFileSystem:%files   | Public files directory is writable.                                              |
| checkFileSystem:%private | Private files directory is writable.                                             |
| checkFileSystem:%temp    | Temporary files directory is writable.                                           |
| checkDbConnection        | Connected to database.                                                           |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                   |
| checkDrupalInstalled     | Drupal is installed.                                                             |
| checkCaching:page        | Drupal cache is disabled.                                                        |
| checkCaching:css         | CSS preprocessing is disabled.                                                   |
| checkCaching:js          | JS preprocessing is disabled.                                                    |
| checkCiConfig            | Git remotes are set in project.yml.                                              |
| checkComposer:require    | acquia/blt is in composer.json's require object.                                 |
| checkBehat:exists        | Behat local settings file exists.                                                |
| checkBehat:root          | You have DrupalVM initialized, but drupal_root in tests/behat/local.yml does not |
|                          | reference the DrupalVM docroot.                                                  |
|                          |   Behat drupal_root is /Users/snize/dev/myproject/docroot.                          |
|                          |   To resolve, run blt setup:behat.                                               |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                              |
| checkAcsfConfig          | BLT settings are included in your pre-settings-php include.                      |
| checkDrupalVm:alias      | drush.aliases.local exists your drush aliases file.                              |
| checkDrupalVm            | Drupal VM is configured correctly.                                               |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                   |
+--------------------------+----------------------------------------------------------------------------------+
```

When I run this command:
```
blt sync:refresh
```

I get the following output:
```
...
> sync
> sync:db
Executing command blt sync:db --define drush.alias=self inside of Drupal VM...
[warning] The xDebug extension is loaded. This will significantly decrease performance.
> setup:settings
[Acquia\Blt\Robo\Tasks\DrushTask] Running /var/www/myproject/vendor/bin/drush cache-clear drush --uri=default --yes && /var/www/myproject/vendor/bin/drush sql-drop --uri=default --yes && /var/www/myproject/vendor/bin/drush sql-sync @myproject.prod @self --structure-tables-key=lightweight --create-db --sanitize --uri=default --yes && /var/www/myproject/vendor/bin/drush cache-clear drush --uri=default --yes in /var/www/myproject/docroot
'drush' cache was cleared.                                                                                                [success]
Do you really want to drop all tables in the database drupal? (y/n): y
The external command could not be executed due to an application error.                                                   [error]
                                 [error]sfully (returned: Permission denied (publickey).
, code: 255)
Error: no database record could be found for source @myproject.prod                                                          [error]
[Acquia\Blt\Robo\Tasks\DrushTask]  Exit code 1  Time 10.279s
[error]
Connection to 127.0.0.1 closed.
> setup:update
> setup:config-import
Executing pre-config-import target hook...
[ExecStack] echo 'No pre-config-import configured.'
...
```

And I expected this to happen:

**Comment by danepowell** [comment]
I know that traditionally we didn't run these commands in the VM, then we tried to run them in the VM and started seeing errors like this, so we took them back out of the VM, and then added them back again in #2070, presumably because we worked around the errors in some way? And then considered reverting them in #2084?

I'll have to investigate this, since the history is obviously a bit complicated and I don't want to keep flip flopping.
